src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,56 @@
+package Duke;
+
+import java.util.Scanner;
+
+/**
+ * Parser that parses input into the program
+ */
+public class Parser {
+    private final Duke duke;
+    private final Ui ui;
+    private final Commands commands;
+
+    public Parser(Duke duke, Ui ui) {
+        this.duke = duke;
+        this.ui = ui;
+        this.commands = new Commands();
+    }
+
+    /**
+     *
+     * @param input that is being read
+     * @throws DukeException when input is invalid
+     */
+    public void parse(String input) throws DukeException{
```

Comment:
Should the error for invalid input (for example, unmark -100) be handled here ?
Also, consider using a switch statement instead